### DISCUSSION TITLE
Is there a way to setup kratos and call it's APIs from backend service without using any UI for login/registration?

### DISCUSSION THREAD

**Discussion by ashwin-pandey** [ory/kratos#3518] [Q&A]
What I want to achieve is to create a service that calls Kratos APIs for identity, but the UI that calls my APIs will have no knowledge of Kratos.

I couldn't find any examples related to this in the documentation.

Any suggestions would be helpful, thanks.

**Comment by vinckr** [comment]
Hello @ashwin-pandey 
It is not a common use case, but I dont see why not. You can just call the API directly - I assume you are building a native/mobile application? 
In any case keep [the difference](https://www.ory.sh/docs/identities/native-browser) in mind!

**Reply by ashwin-pandey** [reply]
Yes, I am able to create login and registration flows using Kratos APIs subsequently in the backend, but my issue is still not resolved, let me rephrase the problem I am facing and what I am trying to achieve.

I want to use Hydra as OIDC which will initiate the login and redirect to the login page, the user enters the credentials (email, password), now, on submit, I am calling Kratos to authenticate the user based on those credentials in the backend (I have figured out this part). Now on a successful response from Kratos, hydra generates the tokens.

Now the problem is that, if I want to update the user's information (traits), as per my understanding, I need the session of the user who wants to update their information (traits) which I am not maintaining because I am only using the success or failure response from Kratos for authenticating the user and moving forward with Hydra for token generation. 

Is there a way to achieve this? Any help would be much appreciated.

**Reply by vinckr** [reply]
I see. 

>I am only using the success or failure response from Kratos for authenticating the user and moving forward with Hydra for token generation.

Can you elaborate a bit on this? Maintaining the session using Ory Kratos would make this fairly trivial as opposed to really complex. By the way did you know you can [Convert Ory Sessions to JSON Web Tokens](https://www.ory.sh/docs/identities/session-to-jwt-cors) now?

**Reply by ashwin-pandey** [reply]
Basically, I am not using Kratos for anything other than authentication after the user enters their credentials. However, I found that I can store the session token/session ID for later use (in my case get or update the user information).

**Reply by vinckr** [reply]
I see. 
Maybe this https://github.com/ory/kratos/issues/1912 would be an easier solution?
